### DISCUSSION TITLE
What exactly does MenuItem.ItemTemplate tries to do?

### DISCUSSION THREAD

**Discussion by rabbitism** [AvaloniaUI/Avalonia#9684] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/14807942/207071814-31482dae-e810-46d8-82df-8a99bb1f595e.png)
I tried to set 
1. `Menu.ItemTemplate`, `MenuItem.ItemTemplate`, 
2. With or without MenuItem as wrapper in the template,
3. DataTemplate/TreeDataTemplate,
4. Place inside DataTemplates/or set ItemTemplate,

All of them not working as expected. Is this ItemTemplate working, if yes, how to set it properly?

**Comment by Gillibald** [comment]
As with all ItemsControls the ItemTemplate is used in situations when Items contains non MenuItem elements. They need to be wrapped by a container when that happens. Your example uses MenuItem elements and therefore no containers are generated.

**Reply by rabbitism** [reply]
@Gillibald Hii, thanks for your reply. I tried both with and without MenuItem wrapper but still not working.

**Comment by maxkatz6** [comment]
ItemTemplate is supposed to be a template of the content control inside of the menu. I.e. template of the "Header" property.
Putting MenuItem inside of the MenuItem.ItemTempalate doesn't make much sense, as it basically will create a MenuItem inside of the MenuItem.Header.

Also, related: https://github.com/AvaloniaUI/Avalonia/issues/3100 https://github.com/AvaloniaUI/Avalonia/issues/6692 https://github.com/AvaloniaUI/Documentation/issues/381

**Comment by rabbitism** [comment]
@maxkatz6 I wrapped that in MenuItem because I mentioned, it doesn't work with or without MenuItem as wrapper in the template. I just wannt give an example. Here is a simpler case. Items is bound to a list of string in VM. 

![image](https://user-images.githubusercontent.com/14807942/207239761-a2f90a2b-1701-4791-86c5-8a03965b638e.png)

In this case I would expect both menu item shows `123`, but it still shows the original bound value.

**Reply by timunie** [reply]
@maxkatz6 I tried to reprduce the issue in ControlCatalog. For me it looks like @rabbitism is right and the `DataTemplate` doesn't get applied. I think there must be anything broken. I'll investigate it further.

**Reply by maxkatz6** [reply]
Yeah, that's weird behavior here. ItemTemplate is used for the submenu items (PART_ItemsPresenter) and nothing is used for the header as a template (PART_HeaderPresenter). Not sure how it should properly work with current API.

**Reply by rabbitism** [reply]
I will convert to issue then.

**Reply by grokys** [reply, marked as answer]
I _think_ this is something that I fixed in #9677. I don't have time to check right now but if you have time to confirm if it's fixed or not that would be nice!
